src/diffusers/training_utils.py

```diff
@@ -431,14 +434,12 @@ def step(self, parameters: Iterable[torch.nn.Parameter]):
         one_minus_decay = 1 - decay
 
         context_manager = contextlib.nullcontext
-        if is_transformers_available() and transformers.integrations.deepspeed.is_deepspeed_zero3_enabled():
-            import deepspeed
 
         if self.foreach:
             if is_transformers_available() and transformers.integrations.deepspeed.is_deepspeed_zero3_enabled():
                 context_manager = deepspeed.zero.GatheredParameters(parameters, modifier_rank=None)
 
-            with context_manager():
+            with context_manager:
```

Comment:
This is correct for GatheredParameters but I think will fail in non-DeepSpeed scenarious where nullcontext is used. Could you create the context_manager object too?

```
context_manager = contextlib.nullcontext()
if deepspeed enabled:
    context_manager = GatheredParameters(...)
```